flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/source/enumerator/TestContinuousSplitPlannerImpl.java

```diff
@@ -0,0 +1,461 @@
+import org.joda.time.format.DateTimeFormat;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestName;
+
+public class TestContinuousSplitPlannerImpl {
+  @ClassRule
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+  private static final FileFormat fileFormat = FileFormat.PARQUET;
+  private static final AtomicLong randomSeed = new AtomicLong();
+
+  @Rule
+  public final HadoopTableResource tableResource = new HadoopTableResource(TEMPORARY_FOLDER,
+      TestFixtures.DATABASE, TestFixtures.TABLE, TestFixtures.SCHEMA);
+
+  @Rule
+  public TestName testName = new TestName();
+
+  private GenericAppenderHelper dataAppender;
+  private DataFile dataFile1;
+  private Snapshot snapshot1;
+  private DataFile dataFile2;
+  private Snapshot snapshot2;
+
+  @Before
+  public void before() throws IOException {
+    dataAppender = new GenericAppenderHelper(tableResource.table(), fileFormat, TEMPORARY_FOLDER);
+  }
+
+  private void appendTwoSnapshots() throws IOException {
+    // snapshot1
+    List<Record> batch1 = RandomGenericData.generate(TestFixtures.SCHEMA, 2, 0L);
+    dataFile1 = dataAppender.writeFile(null, batch1);
+    dataAppender.appendToTable(dataFile1);
+    snapshot1 = tableResource.table().currentSnapshot();
+
+    // snapshot2
+    List<Record> batch2 = RandomGenericData.generate(TestFixtures.SCHEMA, 2, 1L);
+    dataFile2 = dataAppender.writeFile(null, batch2);
+    dataAppender.appendToTable(dataFile2);
+    snapshot2 = tableResource.table().currentSnapshot();
+  }
+
+  /**
+   * @return the last enumerated snapshot id
+   */
+  private IcebergEnumeratorPosition verifyOneCycle(
+      ContinuousSplitPlannerImpl splitPlanner, IcebergEnumeratorPosition lastPosition) throws Exception {
+    List<Record> batch = RandomGenericData.generate(TestFixtures.SCHEMA, 2, randomSeed.incrementAndGet());
+    DataFile dataFile = dataAppender.writeFile(null, batch);
+    dataAppender.appendToTable(dataFile);
+    Snapshot snapshot = tableResource.table().currentSnapshot();
+
+    ContinuousEnumerationResult result = splitPlanner.planSplits(lastPosition);
+    Assert.assertEquals(lastPosition.snapshotId(), result.fromPosition().snapshotId());
+    Assert.assertEquals(lastPosition.snapshotTimestampMs(), result.fromPosition().snapshotTimestampMs());
+    Assert.assertEquals(snapshot.snapshotId(), result.toPosition().snapshotId().longValue());
+    Assert.assertEquals(snapshot.timestampMillis(), result.toPosition().snapshotTimestampMs().longValue());
+    Assert.assertEquals(1, result.splits().size());
+    IcebergSourceSplit split = result.splits().iterator().next();
+    Assert.assertEquals(1, split.task().files().size());
+    Assert.assertEquals(dataFile.path().toString(), split.task().files().iterator().next().file().path().toString());
```

Comment:
Minor: you can use `Iterables.getOnlyElement` to avoid calling `iterator().next()` without validation.